client/lib/mutation-observer/index.js

```diff
@@ -9,7 +9,7 @@ const MutationObserver = ( dataAttributeName, blockBuilder ) => {
 		return blockObserver( dataAttributeName, blockBuilder );
 	}
 
-	window.addEventListener( 'load', () =>
+	document.addEventListener( 'DOMContentLoaded', () =>
```

Comment:
Since this is changing to `DOMContentLoaded` we may find ourselves in an inbetween state where `blockObserver` is never called since the condition just above this is only checking for `'complete' === document.readyState`.

Can you modify it to check for both `complete` and `interactive`?

https://developer.mozilla.org/en-US/docs/Web/API/Document/readyState

Reply:
@jgcaruso code updated to watch either state update